### PR TITLE
ERM29351

### PR DESCRIPTION
**Pull request by Jonty16117**
- The new instance name is passed to the globals and then into the process manager component
- This will only happen when the instance is non root
- Handled the catching this globals data (having instance name) into addInfo inside process manager and can be used there in future.
- However I didn't test the component in the farm setup for now
- Fixed phpcs errors and warnings
- Applied check while setting global variable to []
- Fixed wrong function arguments bug: appended array of strings

### CODE REVIEW

File: src/ManagedProcess.php [hunk 1]
```diff
@@ -26,9 +26,10 @@ public function __construct( array $steps, ?int $timeout = 60 ) {
 	 * @param ProcessManager $manager
 	 * @param array|null $data
 	 * @param string|null $pid ID of the exsting process to continue
+	 * @param string[] $addInfo Additional information can be passed inside the array or [] if none
 	 * @return string ProcessID
 	 */
-	public function start( ProcessManager $manager, $data = [], $pid = null ) {
+	public function start( ProcessManager $manager, $data = [], $pid = null, $addInfo = [] ) {
```
Comment (osnard) [inline src/ManagedProcess.php]: It is not actually "information" that is being passed, but command line arguments for the `maintenance/processExecution.php` script

File: src/ManagedProcess.php [hunk 2]
```diff
@@ -50,8 +51,7 @@ public function start( ProcessManager $manager, $data = [], $pid = null ) {
 		}
 
 		$this->parentProcess = new AsyncProcess( [
-			$phpBinaryPath, $scriptPath, $maintenancePath, $pid
-		] );
+			$phpBinaryPath, $scriptPath, $maintenancePath, $pid ] + $addInfo );
 		$input = new InputStream();
```
Comment (osnard) [inline src/ManagedProcess.php:86]: Unfortunately I have already merged the "structured logging" PR. Could you please log all the arguments that are passed? Also the "input" and "output"

**Review by osnard — CHANGES_REQUESTED** [review]

File: src/ManagedProcess.php (hunk 2 above)
Comment (Jonty16117) [reply src/ManagedProcess.php:86]: eeyup sure... :raised_hands:

**Comment by Jonty16117** [comment]
- Updated variable name and logged start fun
- Resolved merge conflicts

**Review by osnard — APPROVED** [review]

**Pull request merged**